venues/NeurIPS.cc/2021/Conference/python/build-conflicts.py

```diff
@@ -0,0 +1,229 @@
+            edge_nonreaders = {
+                'values': [conference.get_authors_id(number=paper_number)]
+            }
+
+    readers = {
+        'values-copied': edge_readers + ['{tail}']
+    }
+
+    edge_head_type = 'Note'
+    edge_head_query = {
+        'invitation' : conference.get_blind_submission_id()
+    }
+    if 'Custom_Max_Papers' in edge_id:
+        edge_head_type = 'Group'
+        edge_head_query = {
+            'id' : edge_id.split('/-/')[0]
+        }
+    if is_senior_area_chair:
+        edge_head_type = 'Profile'
+        edge_head_query = {
+            'group' : conference.get_area_chairs_id()
+        }
+
+    invitation = openreview.Invitation(
+        id=edge_id,
+        invitees=[conference.get_id(), conference.support_user],
+        readers=[conference.get_id(), conference.get_senior_area_chairs_id(), conference.get_area_chairs_id()],
+        writers=[conference.get_id()],
+        signatures=[conference.get_id()],
+        reply={
+            'readers': readers,
+            'nonreaders': edge_nonreaders,
+            'writers': {
+                'values': edge_writers
+            },
+            'signatures': {
+                'values-regex': '|'.join(edge_signatures),
+                'default': conference.get_program_chairs_id()
+            },
+            'content': {
+                'head': {
+                    'type': edge_head_type,
+                    'query' : edge_head_query
+                },
+                'tail': {
+                    'type': 'Profile',
+                    'query' : {
+                        'group' : match_group.id
+                    }
+                },
+                'weight': {
+                    'value-regex': r'[-+]?[0-9]*\.?[0-9]*'
+                },
+                'label': {
+                    'value-regex': '.*'
+                }
+            }
+        })
+
+    invitation = client.post_invitation(invitation)
+    client.delete_edges(invitation.id)
+    return invitation
+
+
+def build_conflicts(match_group, submissions):
+    edges=[]
+
+    invitation=_create_edge_invitation(conference.get_invitation_id('Conflict',prefix=match_group.id), match_group)
+
+    for submission in submissions:
+
+        author_domains = set()
+        author_emails = set()
+        author_relations = set()
+        author_publications = set()
+
+        author_profiles=openreview.matching._get_profiles(client, submission.details['original']['content']['authorids'])
+        for author in author_profiles:
+            author_info = get_profile_info(author)
+            author_domains.update(author_info['domains'])
+            author_emails.update(author_info['emails'])
+            author_relations.update(author_info['relations'])
+            author_publications.update(author_info['publications'])
+
+        user_profiles=openreview.matching._get_profiles(client, match_group.members)
```

Comment:
I would do this outside of the submissions loop, the profiles list for all the reviewers is always the same, no matter which submission you are checking the conflicts. It will reduce the amount of network calls done in the server.

Reply:
fixed